python/GafferUI/EventLoop.py

```diff
@@ -198,6 +198,25 @@ def removeIdleCallback( cls, callback ) :
 
 		cls.__idleCallbacks.remove( callback )
 
+	## Convenience method to introduce a delay on the mainEventLoop().
+	@classmethod
+	def waitForIdle( cls, count = 1000 ) :
```

Comment:
It's definitely a hack, even if it is necessary. We're basically saying "we have no idea when Qt will have finished doing all its pending updates, so we're just gonna cross our fingers and wait a while". And it seems like the "fix" for the test is just to disconnect the idle counter when it receives the expected number of calls, so it's now ignoring any unexpected ones?

I'd be happier if we could take the guesswork out of this so that it was a legitimate `doAllPendingStuff()`. I just had a go at this using `QApplication.processEvents()` and `QApplication.hasPendingEvents()`, but it seemed to claim to always have pending events, so never returned....